src/lv_widgets/lv_arc.c

```diff
@@ -194,21 +193,16 @@ void lv_arc_set_end_angle(lv_obj_t * arc, int16_t end)
  * @param start the start angle
  * @param end the end angle
  */
-void lv_arc_set_angles(lv_obj_t * arc, int16_t start, int16_t end)
+void lv_arc_set_angles(lv_obj_t * arc, uint16_t start, uint16_t end)
 {
     LV_ASSERT_OBJ(arc, LV_OBJX_NAME);
 
     lv_arc_ext_t * ext = lv_obj_get_ext_attr(arc);
 
-    if(end < 0) end += 360;
-
```

Comment:
I've intentionally skipped `if(end > 360) end -= 360;` because if I set e.g. `start = 90`, `end=360+90` I would expect to see a full circle. However by limiting `end` this information is lost and no arc will be drawn if both `start` and `end` are 90.  

I've used `+-360` instead of `% 360` because I thought it's faster. What do think about it?

Reply:
All the < 0 comparisons are not necessary now since the type is unsigned.

But I see the problem using modulus which will not allow a complete circle.  To some degree adding a rotation function fixes this but of course we want existing code to work.

A quick check for the cortex M4 shows that modulus takes 6-16 cycles where are a comparison and subtraction takes 4-7 cycles.

I'll revise and add the rotation function.

Reply:
I've made the changes but still have a redraw bug when both start and end are > 360, which can happen with rotation.  Out of time today so I'll look again when I can.

Reply:
I limited the end values to the start values + 360.  It will not invalidate correctly if start=360 and end = 0, but I going to leave it as it is.

Reply:
> I limited the end values to the start values + 360. It will not invalidate correctly if start=360 and end = 0, but I going to leave it as it is.

That's fine. I can't imagine what would be expected behavior with start=360 and end = 0.